app/controllers/users_controller.rb

```diff
@@ -0,0 +1,72 @@
+class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user,   only: [:edit, :update]
+  before_action :admin_user,     only: :destroy
+
+  def index
```

Comment:
the order of the crud methods should be: `index, show, new, create, edit, update, destroy`